demos/minimap/minimap.js

```diff
@@ -0,0 +1,305 @@
+* @param {Workspace} minimap The workspace that will be used as a minimap.
+*/
+Minimap.init = function(workspace, minimap){
+  this.workspace = workspace;
+  this.minimap = minimap;
+
+  //Adding scroll callback functionlity to vScroll and hScroll just for this demo.
+  //IMPORTANT: This should be changed when there is proper UI event handling
+  //           api available and should be handled by workspace's event listeners.
+  this.workspace.scrollbar.vScroll.setHandlePosition = function(newPosition){
+    this.handlePosition_ = newPosition;
+    this.svgHandle_.setAttribute(this.positionAttribute_, this.handlePosition_);
+
+    // Code above is same as the original setHandlePosition function in core/scrollbar.js.
+    // New code starts from here.
+
+    // Get the absolutePosition.
+    var absolutePosition = (this.handlePosition_ / this.ratio_);
+
+    // Firing the scroll change listener.
+    Minimap.onScrollChange(absolutePosition, this.horizontal_);
+  };
+
+  // Adding call back for horizontal scroll.
+  this.workspace.scrollbar.hScroll.setHandlePosition = function(newPosition){
+    this.handlePosition_ = newPosition;
+    this.svgHandle_.setAttribute(this.positionAttribute_, this.handlePosition_);
+
+    // Code above is same as the original setHandlePosition function in core/scrollbar.js.
+    // New code starts from here.
+
+    // Get the absolutePosition.
+    var absolutePosition = (this.handlePosition_ / this.ratio_);
+
+    // Firing the scroll change listener.
+    Minimap.onScrollChange(absolutePosition, this.horizontal_);
+  };
+
+
+  // Required to stop a positive feedback loop when user clicks minimap
+  // and the scroll changes, which inturn may change minimap.
+  this.disableScrollChange = false;
+
+  // Listen to events on the main workspace.
+  this.workspace.addChangeListener(Minimap.mirrorEvent);
+
+  //Get rectangle bounding the minimap div.
+  this.rect = document.getElementById('mapDiv').getBoundingClientRect();
+
+  // Create a svg overlay on the top of mapDiv for the minimap.
+  this.svg = Blockly.utils.createSvgElement('svg', {
+    'xmlns': 'http://www.w3.org/2000/svg',
+    'xmlns:html': 'http://www.w3.org/1999/xhtml',
+    'xmlns:xlink': 'http://www.w3.org/1999/xlink',
+    'version': '1.1',
+    'height': this.rect.bottom-this.rect.top,
+    'width': this.rect.right-this.rect.left,
+    'class': 'minimap',
+  }, document.getElementById('mapDiv'));
+  this.svg.style.top = this.rect.top;
+  this.svg.style.left = this.rect.left;
+
+  // Creating a rectangle in the minimap that represents current view.
+  Blockly.utils.createSvgElement('rect', {
+    'width':100,
+    'height':100,
+    'class':'mapDragger'
+  }, this.svg);
+
+  // Rectangle in the minimap that represents current view.
+  this.mapDragger = this.svg.childNodes[0];
+
+  // Adding mouse events to the rectangle, to make it Draggable.
+  this.mapDragger.addEventListener("mousedown", Minimap.mousedown,false);
```

Comment:
Right now minimap dragging breaks if you start a drag inside the box, drag outside, and then drag back inside.  You're catching the mouseout and stopping the drag, but it's not obvious to the user what's going on.  

Two things to do to fix problems here:
- register the mousedown listener here and only register the other listeners (mousemove, mouseup) in the mousedown listener.  That way those listeners only exist during a drag.  Unregister them on mouse up.
- Register your mouse move listener on the whole document instead of on the minimap.  Then it doesn't matter if they drag outside of the minimap.

Then you don't need the mouseout listener at all.